test/e2e/apimachinery/garbage_collector.go

```diff
@@ -605,24 +607,28 @@ var _ = SIGDescribe("Garbage collector", func() {
 		if err := deployClient.Delete(deployment.ObjectMeta.Name, deleteOptions); err != nil {
 			framework.Failf("failed to delete the deployment: %v", err)
 		}
-		By("wait for 2 Minute to see if the garbage collector mistakenly deletes the rs")
-		err = wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
-			return verifyRemainingDeploymentsReplicaSetsPods(f, clientSet, deployment, 0, 1, 2)
-		})
+		By("wait for 10s to see if the garbage collector mistakenly deletes the rs")
```

Comment:
This is changed because it doesn't make sense to use PollImmediate to make sure something _doesn't_ happen. If it hasn't happened by the first polling the PollImmediate will succeed and then what we are watching for may end up occurring later and we will miss it.

Reply:
Good catch. Could you keep the code consistent with https://github.com/kubernetes/kubernetes/blame/6546b69964f96ff6dd7bfdd8e72f89f09bd3e305/test/e2e/apimachinery/garbage_collector.go#L447-L459?

Reply:
Done